flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -387,28 +368,29 @@ class ScalarExprLowering {
   template <int KIND>
   fir::ExtendedValue
   genval(const Fortran::evaluate::ComplexComponent<KIND> &part) {
-    auto lhs = genunbox(part.left());
-    assert(lhs && "unexpected array expression complex component");
-    return extractComplexPart(lhs, part.isImaginaryPart);
+    return extractComplexPart(genunbox(part.left()), part.isImaginaryPart);
```

Comment:
Should `genunbox` crash if this is called on something that is not an unboxed value ?
I understand that now that array have a different path, these asserts/TODO should not be hit, but I am a bit worried that we may end up in weird segfaults with compiler bugs. In non debug mode, working on empty `mlir::value` does not crash nicely.

Reply:
It does.

Reply:
Thanks, I saw your addition of `fir::emitFatalError` inside it now. Sorry for bothering you for nothing about it !